api/src/main/java/marquez/db/DatasetDao.java

```diff
@@ -272,6 +272,18 @@ DatasetRow upsert(
       """)
   Optional<DatasetRow> delete(String namespaceName, String name);
 
+  @SqlUpdate(
+      """
+        DELETE FROM datasets_tag_mapping dtm
+        WHERE EXISTS (
+            SELECT 1
+            FROM datasets d
+            JOIN tags t ON d.uuid = dtm.dataset_uuid AND t.uuid = dtm.tag_uuid
+            WHERE d.name = :datasetName AND t.name = :tagName
```

Comment:
Shouldn't where condition contain namespace filter as well?

Reply:
Not sure it would add much given the data model of 

datasets -> datasets_tag_mapping <- tags

Suspect when the tag linkage is created in the first place that the dataset_uuid is added based on the namespace provided so not sure it adds much in this case but happy to put it in.

Totally agree on the integration tests though @wslulciuc has been really helpful with that.